client/cmdhfmf.c

```diff
@@ -1286,15 +1306,17 @@ int CmdHF14AMfChk(const char *Cmd)
 				for (uint16_t t = 0; t < 2; t++) {
 					if (e_sector[sectorNo].foundKey[t]) {
 						num_to_bytes(e_sector[sectorNo].Key[t], 6, block + t * 10);
+                        keyFoundCount++; // Key found count for information
 					}
 				}
 				mfEmlSetMem(block, FirstBlockOfSector(sectorNo) + NumBlocksPerSector(sectorNo) - 1, 1);
 			}
 		}
-		PrintAndLog("Found keys have been transferred to the emulator memory");
+		// Updated to show the actual number of keys found/transfered.
+		PrintAndLog("%d keys(s) found have been transferred to the emulator memory",keyFoundCount);
 	}
 
-	if (createDumpFile) {
+	if (createDumpFile && !singleBlock) {

```

Comment:
Shouldn't we add some `PrintAndLog()` for the `singleBlock` case? To inform the user that nothing has been dumped? Or should be we even prevent this combination of options by checking the parameters and throw and error if the `d` option is used with a single block number?

Reply:
I added a log message and exit.  Happy to change the wording if needed.